tools/clang-tidy.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# Copyright 2023 Ekumen, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Run clang-tidy for static analysis in the Beluga repository.
+
+[[ -z "${WITHIN_DEV}" ]] && echo -e "\033[1;33mWARNING: Try running this script inside the development container if you experience any issues.\033[0m"
+
+set -o errexit -o xtrace
+
+ROS_PACKAGES="beluga beluga_system_tests"
+
+source /opt/ros/${ROS_DISTRO}/setup.sh
+colcon build --packages-up-to ${ROS_PACKAGES} --event-handlers=console_cohesion+ --symlink-install --mixin ccache
+echo ${ROS_PACKAGES} | xargs -n1 echo | xargs -I{} run-clang-tidy -quiet -j 4 -p ./build/{}
```

Comment:
@nahueespinosa meta: have you consider [hooking up `clang-tidy` to `cmake`](https://www.danielsieger.com/blog/2021/12/21/clang-tidy-cmake.html)?

Reply:
Done in https://github.com/Ekumen-OS/beluga/pull/197/commits/471ac152cc139961331d9c0cf4a49a35e807aec9.

Reply:
I don't know about this one, I haven't actually looked into that yet but:
- [`run-clang-tidy`](https://clang.llvm.org/extra/doxygen/run-clang-tidy_8py_source.html) can run in parallel.
- We have a repository-level `.clang-tidy` config file, I guess we can duplicate it for each package.

I'll take a look in the following days.

Reply:
I think I prefer keeping the `clang-tidy` config file at the root of the repository for consistency with `clang-format`. High level thoughts are:
- We build, run tests, generate coverage reports, and generate documentation per package.
- We lint and run static analysis tools at a repository level (`pre-commit`, `run-clang-tidy`, etc.).
- We don't run static analysis on pre-commit hooks for the reasons stated in #147.